aws/resource_aws_connect_instance_test.go

```diff
@@ -0,0 +1,346 @@
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).connectconn
+	ctx := context.Background()
+	// MaxResults:  Maximum value of 10. https://docs.aws.amazon.com/connect/latest/APIReference/API_ListInstances.html
+	input := &connect.ListInstancesInput{MaxResults: aws.Int64(10)}
+	var sweeperErrs *multierror.Error
+	for {
+		listOutput, err := conn.ListInstances(input)
+		if err != nil {
+			if testSweepSkipSweepError(err) {
+				log.Printf("[WARN] Skipping Connect Instance sweep for %s: %s", region, err)
+				return nil
+			}
+			return fmt.Errorf("Error retrieving Connect Instance: %s", err)
+		}
+		for _, instance := range listOutput.InstanceSummaryList {
+			id := aws.StringValue(instance.Id)
+			r := resourceAwsConnectInstance()
+			d := r.Data(nil)
+			d.SetId(id)
+
+			diags := r.DeleteContext(ctx, d, client)
+			for i := range diags {
+				if diags[i].Severity == diag.Error {
+					log.Printf("[ERROR] %s", diags[i].Summary)
+					sweeperErrs = multierror.Append(sweeperErrs, fmt.Errorf(diags[i].Summary))
+					continue
+				}
+			}
+		}
+		if aws.StringValue(listOutput.NextToken) == "" {
+			break
+		}
+		input.NextToken = listOutput.NextToken
+	}
+	return sweeperErrs.ErrorOrNil()
+}
+func TestAccAwsConnectInstance_basic(t *testing.T) {
+	var v connect.DescribeInstanceOutput
+	rInt := acctest.RandInt()
+	resourceName := "aws_connect_instance.foo"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsConnectInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsConnectInstanceConfigBasic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsConnectInstanceExists(resourceName, &v),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
+					resource.TestCheckResourceAttrSet(resourceName, "created_time"),
+					resource.TestCheckResourceAttr(resourceName, "identity_management_type", connect.DirectoryTypeConnectManaged),
+					resource.TestCheckResourceAttrSet(resourceName, "instance_alias"),
+					resource.TestCheckResourceAttr(resourceName, "inbound_calls_enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "outbound_calls_enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "contact_flow_logs_enabled", "false"),
+					resource.TestCheckResourceAttr(resourceName, "contact_lens_enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "auto_resolve_best_voices", "true"),
+					resource.TestCheckResourceAttr(resourceName, "use_custom_tts_voices", "false"),
+					resource.TestCheckResourceAttr(resourceName, "early_media_enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "status", connect.InstanceStatusActive),
+					resource.TestCheckResourceAttrSet(resourceName, "service_role"),
```

Comment:
Any validation we can do here? what is returned? if its the ARN of the role see the ARN check comment above